include/unifex/task.hpp

```diff
@@ -282,6 +298,15 @@ struct _task<T>::type {
 template <typename T>
 using task = typename _task::_task<T>::type;
 
+// Await this to cancel and unwind if stop has been requested:
+inline constexpr _task::_stop_if_requested stop_if_requested {};
+
+// Await this to cancel unconditionally:
+inline constexpr auto stop = just_done();
```

Comment:
This is creating a separate dialect for coroutines.
Why `co_await stop` and not `co_await just_done();`?

Reply:
I'm ambivalent. On the one hand, I'm in favor of keeping the library small, and not creating dialects. On the other hand, with "done" and "stop" we already have two ways of saying the same thing. We have `set_done`, `just_done`, and `transform_done`, but we also have `inplace_stop_[token|source]`, `request_stop`, and `stop_requested`. I expect the "stop"-flavored names to be used extensively in coroutines (though some in sender/receiver), and the "done"-flavored names to be used primarily in sender/receiver.

When you have a codebase filled with lots of coroutines and little to no sender/receiver, `co_await just_done()` doesn't roll of the tongue.

All that said, I'm happy to leave it out and define it outside libunifex for the convenience of the clients of `unifex::task<>`.